src/dpos-user.ts

```diff
@@ -379,20 +397,62 @@ export class DPOSUser {
     }
     signature = pendingReceipt.oracleSignature
 
+    log('Got receipt', pendingReceipt)
     return CryptoUtils.bytesToHexAddr(signature)
   }
 
   private async withdrawCoinFromRinkebyGatewayAsync(
     amount: BN,
     sig: string
   ): Promise<ethers.ContractTransaction> {
+    const hash = await this.createWithdrawalHash(amount)
+    log('Receipt hash:', hash)
+
+    if (this._version === 2) {
+      // Ugly hack to extract the 'mode' bit from the old signature format - if it's still used (68 = 66 + 2, where 2 is the 0x)
+      sig = sig.length === 134 ? '0x' + sig.slice(4) : sig
+      let sign = ethers.utils.splitSignature(sig)
+
+      let signer = ethers.utils.recoverAddress(
+        ethers.utils.arrayify(ethers.utils.hashMessage(ethers.utils.arrayify(hash))),
+        sign
+      )
+      log('Receipt was signed by:', signer)
+
+      let valIndexes = [0]
```

Comment:
Note that in a future PR this was edited to be calculated based on the current validators list from the ValidatorManagerContract. The current task does not require it yet so I prefer to get this in as is, and complete it later. For reference, this is what's going to happen: https://github.com/loomnetwork/loom-js/pull/216/commits/ee0b806f29b44002516a0d58df9c9c07a11cd2b5